### ISSUE TITLE
Some tables do not open anymore. OData.error

### ISSUE THREAD

**Issue by dhagarw** [microsoft/AzureStorageExplorer#5407]
### Preflight Checklist

- [X] I have installed the [latest version of Storage Explorer](https://github.com/Microsoft/AzureStorageExplorer/releases/latest).
- [X] I have checked existing resources, including the [troubleshooting guide](https://docs.microsoft.com/azure/storage/common/storage-explorer-troubleshooting) and the [release notes](https://github.com/microsoft/AzureStorageExplorer/releases).
- [X] I have searched for [similar issues](https://github.com/Microsoft/AzureStorageExplorer/issues).

### Storage Explorer Version

1.22.1

### Regression From

1.14.12

### Architecture

i86

### Storage Explorer Build Number

20220122.3

### Platform

Windows

### OS Version

windows 11

### Bug Description

I am unable to open a specific table anymore. I can still open it from older versions.

I have pasted the screen shot of the error. 

![image](https://user-images.githubusercontent.com/99184395/152786103-3dd87fca-c9e4-43ed-b278-878bc5a978da.png)

### Steps to Reproduce

Hard to reproduce. It is happening for a specific table.

### Actual Experience

The table doesnot open. 

![image](https://user-images.githubusercontent.com/99184395/152786272-f0f50434-669a-4f61-ae8a-588fc2a237d6.png)

### Expected Experience

The table should open and have data in it.

**Comment by MRayermannMSFT** [comment]
@dhagarw

- How are you accessing this table? Sign in? Attach with SAS? Attach with key?
- Can you go to Help -> Open Logs Directory, go to the newest folder, look for a file with `table_extension` in the name, and then upload that file here?
- Would you be comfortable sharing a SAS token to the table with us? It can be read only. We can give you an e mail address that you can send it to so you don't have to post it publicly.

**Comment by vaishnaviithamraju** [comment]
I was facing the same issue. I was unable to open some tables that i used to access earlier using SAS with a key. Downgrading the storage explorer to v1.21.3 did the trick. It works fine now.

**Comment by dhagarw** [comment]
> 
Hi @MRayermannMSFT 
I am accessing this table via Attach with Key.

Attaching the log file.
[2022-02-21_163425_se-table-extension_22512.log](https://github.com/microsoft/AzureStorageExplorer/files/8109198/2022-02-21_163425_se-table-extension_22512.log)

Yes I mean 1.14.2 . Sorry for the confusion. Actually I can open the table from 1.14.2 .

I was able to sucessfully open the table from 1.21.3.

Can you share the mail id where I can share the sas key?

**Comment by MRayermannMSFT** [comment]
@dhagarw you can send it to sehelp at microsoft dot com

**Comment by craxal** [comment]
@dhagarw Thanks for the SAS. I'm able to repro your issue. Investigating to see what might be the cause.

One thing I'm noticing is the first query request returns with 0 results and a continuation token. That's a bit unusual, as normally the first requests comes with a few results.

**Comment by craxal** [comment]
@dhagarw On further inspection, I suspect the issue is the result of receiving a bad continuation token.

Normally, the initial query request returns a number of entities along with a continuation token like this:

```
eyJuZXh0UGFydGl0aW9uS2V5IjoiMSE0OCFNV1k0TTJGa01tVXRZamN5WWkwME5URmlMVGd3TlRjdE5XWm1PR0k0T1dRNE9ESXgiLCJuZXh0Um93S2V5IjoiMSE0OCFOVGN3TnpsbVlUWXRPRGt6TUMwME0yVTFMV0k1TUdRdE5qUTNOall6WVRsalpEWmoifQ==
```

This is a Base64 encoded string, which looks like this when decoded:
```json
{
  "nextPartitionKey": "1!48!MWY4M2FkMmUtYjcyYi00NTFiLTgwNTctNWZmOGI4OWQ4ODIx",
  "nextRowKey": "1!48!NTcwNzlmYTYtODkzMC00M2U1LWI5MGQtNjQ3NjYzYTljZDZj"
}
```

However, the initial query for the table you're having trouble with returns with zero results and a continuation that looks like this:
```
eyJuZXh0UGFydGl0aW9uS2V5IjoiMSEwISIsIm5leHRSb3dLZXkiOiIifQ==
```

If we decode that:
```json
{
  "nextPartitionKey": "1!0!",
  "nextRowKey": ""
}
```

Notice it's a lot shorter, and the `nextRowKey` has en empty string. I believe that's where the "invalid input" error message is coming from.

The odd continuation token data comes straight from the HTTP response, by the look of it, which suggests this may be a problem in the service somewhere. We'll get in touch with someone on the service team. In the meantime, it may be worth opening a support ticket with Azure. Be sure to share any request IDs with them (you can get that by examining the log files containing "se-table-extension" in the name and looking for error messages).

**Comment by paulspinksthomsonreuters** [comment]
I am also facing this issue on one of our tables. I have temporarily rolled back to 1.21.0 and can once again access my table.

**Comment by craxal** [comment]
@dhagarw @paulspinksthomsonreuters We recently released 1.23.0. Can you try using that version and tell us if you still encounter the problem?

**Comment by craxal** [comment]
@paulspinkstr Are you able to get a Fiddler trace of the network requests/responses to the table? Do the requests/responses seem to fit the symptoms I detailed above? Can you share your log files?

**Comment by paulspinkstr** [comment]
@craxal, I've tried to get the logs for you (I can't install fiddler on my machine). Frustratingly I can't reproduce this today. 

I've gone back through my logs and I know that I definitely had this problem on March 8th, because that's when I found this thread and I have comms with someone about it.

I have this in my log: 

GET https://<removed>.table.core.windows.net/Case()?NextPartitionKey=1%218%21Q2FzZQ--&NextRowKey=&sv=2020-08-04&ss=bfqt&srt=sco&sp=rltfx&se=2022-03-08T18%3A15%3A21Z&st=2022-03-08T10%3A15%3A21Z&spr=https&sig=AzureSAS Token Redacted - 400
Credential used: SAS token - sv=2020-08-04&ss=bfqt&srt=sco&sp=rltfx&se=2022-03-08T18%3A15%3A21Z&st=2022-03-08T10%3A15%3A21Z&spr=https&sig=******
RequestId: 2d63a29d-86a5-43f4-aefb-52bbc395fcf8, Date sent: Tue, 08 Mar 2022 10:17:54 GMT, ApiVersion: 2019-02-02, ProxySettings: undefined
ErrorCode: InvalidInput

**Comment by craxal** [comment]
@paulspinkstr Do you have another machine you can install Fiddler on? Unless your table contains sensitive information, would you be willing to share a readonly SAS URI with us to help us dig into this issue? Or would you be willing to have a Teams call for a debugging session? You can get in touch with us via email at sehelp @ microsoft.com.

**Comment by craxal** [comment]
Hey everyone, I think we've narrowed the problem to an issue in the table SDK we're using. The SDK appears to be using a default empty string value for NextRowKey, and is putting this in request URLs. This results in the "invalid parameter" error. The reason this only seems to happen for certain tables may be related to partition boundaries. This would also explain why older versions of Storage Explorer (which used a different version of the SDK), do not have this problem.

I have opened https://github.com/Azure/azure-sdk-for-js/issues/20913 and asked a fix take high priority.
